src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/GetAzureContainerRegistry.cs

```diff
@@ -14,65 +14,87 @@
-            HelpMessage = "Container Registry Name.")]
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = RegistryNameParameterSet, HelpMessage = "Container Registry Name.")]
         [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Show more details about the container registry.")]
+        public SwitchParameter IncludeDetail { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The container registry resource id")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ResourceIdAlias)]
+        public string ResourceId { get; set; }
+
         public override void ExecuteCmdlet()
         {
-            if (!string.IsNullOrEmpty(ResourceGroupName) && !string.IsNullOrEmpty(Name))
+            if (string.Equals(ParameterSetName, ListRegistriesParameterSet))
             {
-                var registry = RegistryClient.GetRegistry(ResourceGroupName, Name);
-                WriteObject(new PSContainerRegistry(registry));
+                ListRegistry();
             }
             else
             {
-                List<PSContainerRegistry> list = new List<PSContainerRegistry>();
-
-                var registries = RegistryClient.ListRegistries(ResourceGroupName);
-                foreach (Registry registry in registries)
+                if (MyInvocation.BoundParameters.ContainsKey("ResourceId") || !string.IsNullOrWhiteSpace(ResourceId))
                 {
-                    list.Add(new PSContainerRegistry(registry));
+                    string resourceGroup, registryName, childResourceName;
+                    if(!ConversionUtilities.TryParseRegistryRelatedResourceId(ResourceId, out resourceGroup, out registryName, out childResourceName))
+                    {
+                        WriteInvalidResourceIdError(InvalidRegistryResourceIdErrorMessage);
+                        return;
+                    }
+
+                    ResourceGroupName = resourceGroup;
+                    Name = registryName;
                 }
+                ShowRegistry();
+            }
+        }
+
+        private void ShowRegistry()
+        {
+            var registry = RegistryClient.GetRegistry(ResourceGroupName, Name);
+            var psRegistry = new PSContainerRegistry(registry);
+
+            if (IncludeDetail)
+            {
+                SetRegistryDetials(psRegistry);
+            }
+            WriteObject(psRegistry);
+        }
 
-                while (!string.IsNullOrEmpty(registries.NextPageLink))
+        private void ListRegistry()
+        {
+            var psRegistries = RegistryClient.ListAllRegistries(ResourceGroupName);
+
+            if (IncludeDetail)
+            {
+                foreach(var psr in psRegistries)
```

Comment:
There might be some perf concern here. `list replications` itself is a fanout operation and it is on top of another fanout operation `list registries`. From the modeling with ARM and consistency with Az CLI and Azure Portal, `replication` is an independent resource rather than some detail of registry (`usage` is exactly registry detail).

Reply:
Now only includes usage info.